src/create-sortable.ts

```diff
@@ -1,25 +1,36 @@
-import { createComputed } from "solid-js";
-
 import { createDraggable } from "./create-draggable";
 import { createDroppable } from "./create-droppable";
-import { combineRefs } from "./combine-refs";
+import { RefSetter, combineRefs } from "./combine-refs";
 import { useSortableContext } from "./sortable-context";
-import { useDragDropContext } from "./drag-drop-context";
-import { noopTransform } from "./layout";
+import { Listeners, useDragDropContext } from "./drag-drop-context";
+import { Layout, noopTransform, Transform } from "./layout";
+import { createComputed } from "solid-js";
 
-export const createSortable = (options) => {
-  const { id } = options;
-  const [dndState, { anyDraggableActive, displace }] = useDragDropContext();
-  const [sortableState] = useSortableContext();
-  const draggable = createDraggable(options);
-  const droppable = createDroppable(options);
+interface Sortable {
+  (element: HTMLElement): void;
+  ref: RefSetter<HTMLElement | null>;
+  get transform(): Transform;
+  get dragActivators(): Listeners;
+  get isActiveDraggable(): boolean;
+  get isActiveDroppable(): boolean;
+}
+
+const createSortable = (
+  id: string,
+  data: Record<string, any> = {}
+): Sortable => {
+  const [dndState, { anyDraggableActive, displace }] = useDragDropContext()!;
```

Comment:
[Non-null assertion](https://www.typescriptlang.org/docs/handbook/2/everyday-types.html#non-null-assertion-operator-postfix-) - removes `undefined` and `null` from the type. I'm not keen on wrapping the entire function in an `if` statement and since the context is required for the library to work (as stated in the docs) its presence is assumed.